api/test/unit/controllers/test_search_controller.py

```diff
@@ -13,36 +13,51 @@
 
 
 @pytest.mark.parametrize(
-    "total_hits, real_result_count, page_size, expected",
+    "total_hits, real_result_count, page_size, page, expected",
     [
         # No results
-        (0, 0, 10, (0, 0)),
+        (0, 0, 10, 0, (0, 0)),
         # Setting page size to 0 raises an exception
         pytest.param(
-            0, 0, 0, (0, 0), marks=pytest.mark.raises(exception=ZeroDivisionError)
+            0, 0, 0, 0, (0, 0), marks=pytest.mark.raises(exception=ZeroDivisionError)
```

Comment:
In practice, this function can never be called with 0 as the `page_size` value because we have a validator on `page_size` with a minimum value of 1.

Reply:
Correct, but unit tests generally are not meant to embed meta-knowledge of the application like that. The list here tests the unit's possible (relevant and irrelevant to the application) inputs which makes it easier to verify the intended behaviour (at least in my experience).